pkg/operator/client/deploy.go

```diff
@@ -390,80 +417,63 @@ func (c *Client) uninstallWithHelm(helmDir string, kotsCharts []v1beta1.HelmChar
 				continue
 			}
 			logger.Errorf("error: %s", err.Error())
-			return errors.Wrapf(err, "failed to uninstall release %s for chart %s: %s", dir.ReleaseName, dir.ChartName, stderr)
+			return errors.Wrapf(err, "failed to uninstall release %s for chart %s: %s", chart.GetReleaseName(), chart.GetChartName(), stderr)
 		}
 	}
 
 	return nil
 }
 
-func getRemovedCharts(prevDir string, curDir string, previousKotsCharts []v1beta1.HelmChart, curKotsCharts []v1beta1.HelmChart) ([]v1beta1.HelmChart, error) {
-	if prevDir == "" {
-		return []v1beta1.HelmChart{}, nil
-	}
+// getRemovedCharts returns a list of helm release names that were removed in the current version
+func getRemovedCharts(prevV1Beta1Dir, curV1Beta1Dir string, previousV1Beta1KotsCharts, currentV1Beta1KotsCharts []kotsutil.HelmChartInterface, prevV1Beta2Dir, curV1Beta2Dir string, previousV1Beta2KotsCharts, currentV1Beta2KotsCharts []kotsutil.HelmChartInterface) ([]kotsutil.HelmChartInterface, error) {
```

Comment:
let's move all these parameters to a new struct. something like `GetRemovedChartsOptions`

Reply:
good idea :)